---
title: "MWA-pipeline"
author: "Jennifer Sandoval"
date: "21/07/2022"
---

```{bash}
#Creating a new enviroment for conda for Metagenommic Wide Analysis
conda create -n mwa
conda activate mwa

#Instaling softwares
conda install -c bioconda fastqc
conda install -c bioconda trimmomatic
conda install -c bioconda megahit
conda install -c bioconda bowtie2
conda install -c bioconda samtools
conda install -c bioconda trim-galore

```

```{bash}

#List of metagenomes codes
codes=(ERR1398068 ERR1398168 ERR1398221 ERR1398076 ERR1398077 ERR1398085 ERR1398129 ERR1398206 ERR1398263 ERR1398078 ERR1398257 ERR1398089)
printf "%s\n" "${codes[@]}" > list_metagenomes.txt

#Read the list of codes metagenomes
metagenomes_codes=(`cat "list_metagenomes.txt"`)

for sample in ${metagenomes_codes[*]}; do
    echo $sample
	mkdir $sample
    #Quality control and trimming samples
    trim_galore --paired ./samples/${sample}_1.fastq.gz ./samples/${sample}_2.fastq.gz -o ./$sample;

    #Metagenome assembly 
    cd $sample
    megahit -1 ./${sample}_1_val_1.fq.gz -2 ${sample}_2_val_2.fq.gz -t 8 -o ${sample}_assembly --tmp-dir /tmp

    #Taxonomic profiling using kraken
    kraken --db ../minikraken_20171013_4GB/ ./${sample}_assembly/final.contigs.fasta --output ./kraken/${sample}_kraken.out --threads 8

    #Parse kraken output file
    cut -f2,3 ./kraken/${sample}_kraken.out > ./kraken/${sample}_kraken.krona.in
    #Generating krona visualization for taxonomic profiles
    ktImportTaxonomy ./kraken/${sample}_kraken.krona.in -o ./kraken/${sample}_krona.html

    #Generating kraken report
    kraken-report --db ../minikraken_20171013_4GB/ ./kraken/${sample}_kraken.out > ./kraken/${sample}_kraken.rpt
done


#Functional annotation and relative abundances
superfocus -q ${codes[0]}/${codes[0]}_assembly/final.contigs.fasta -q ${codes[1]}/${codes[1]}_assembly/final.contigs.fasta -q ${codes[2]}/${codes[2]}_assembly/final.contigs.fasta -q ${codes[3]}/${codes[3]}_assembly/final.contigs.fasta  -q ${codes[4]}/${codes[4]}_assembly/final.contigs.fasta -q ${codes[5]}/${codes[5]}_assembly/final.contigs.fasta -q ${codes[6]}/${codes[6]}_assembly/final.contigs.fasta -q ${codes[7]}/${codes[7]}_assembly/final.contigs.fasta -q ${codes[8]}/${codes[8]}_assembly/final.contigs.fasta -q ${codes[9]}/${codes[9]}_assembly/final.contigs.fasta -q ${codes[10]}/${codes[10]}_assembly/final.contigs.fasta -q ${codes[11]}/${codes[11]}_assembly/final.contigs.fasta  -dir general_output_focus_v2 -a diamond -t 8 -n 1


#Generating a taxonomic profile report for 
kraken-mpa-report --db minikraken_20171013_4GB/ ./${codes[0]}/kraken/${codes[0]}_kraken.out ./${codes[1]}/kraken/${codes[1]}_kraken.out ./${codes[2]}/kraken/${codes[2]}_kraken.out ./${codes[3]}/kraken/${codes[3]}_kraken.out ./${codes[4]}/kraken/${codes[4]}_kraken.out ./${codes[5]}/kraken/${codes[5]}_kraken.out ./${codes[6]}/kraken/${codes[6]}_kraken.out ./${codes[7]}/kraken/${codes[7]}_kraken.out ./${codes[7]}/kraken/${codes[7]}_kraken.out ./${codes[8]}/kraken/${codes[8]}_kraken.out ./${codes[9]}/kraken/${codes[9]}_kraken.out ./${codes[10]}/kraken/${codes[10]}_kraken.out ./${codes[11]}/kraken/${codes[11]}_kraken.out> all_samples_report.txt



```


